daemon/algod/api/server/auth.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package server
+
+import (
+	"crypto/subtle"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+
+	"github.com/algorand/go-algorand/daemon/algod/api/server/lib/middlewares"
+	"github.com/algorand/go-algorand/logging"
+)
+
+const urlAuthFormatter = "/urlAuth/%s"
+const debugRouteName = "debug"
+
+// AuthRoutes define the desired mapping of authentication token keys to allowed routes.
+type AuthRoutes map[string]map[echo.Route]echo.HandlerFunc
```

Comment:
The user-facing implementation has two keys, that's true. But the inner handler is capable to handle as many as defined. 
If we'll decide to extend the user facing interface to support more sophisticated schema ( i.e. json db ), we wouldn't need to change this file, but rather just the "configuration" part. 
I tried to follow the spirit of https://github.com/algorand/go-algorand/issues/863, supporting any number of tokens, rather than forcing it into two.